Blog post: Insights from the OpenTelemetry Docs Usability Survey (#5797)
open-telemetry/opentelemetry.io: content/en/blog/2024/otel-docs-survey/index.md

---
title: Insights from the OpenTelemetry Docs Usability Survey

> linkTitle: OpenTelemetry Docs Survey

Reviewer: ```suggestion
linkTitle: OTel-docs Survey
```

date: 2024-12-13
author: '[Tiffany Hrabusa](https://github.com/tiffany76) (Grafana Labs)'
issue: https://github.com/open-telemetry/opentelemetry.io/issues/5793
sig: Communications, End-User

> # prettier-ignore

Reviewer: ```suggestion
```

cSpell:ignore: Hrabusa
---

> The OpenTelemetry End-User SIG recently surveyed the community to find out how

Reviewer 2: ```suggestion
[The OpenTelemetry End-User SIG](/community/end-user/) recently surveyed the community to find out how
```

> user-friendly OpenTelemetry’s documentation is. In an earlier survey, two-thirds

Reviewer 2: ```suggestion
user-friendly [OpenTelemetry’s documentation](/docs/) is. In an earlier survey, two-thirds
```

of respondents named comprehensive documentation as a top resource they wished
they’d had when getting started with OpenTelemetry. So we decided to dig a
little deeper. The Docs Usability Survey asked users where they go for OTel
documentation, what they’d like to see more of in the docs, and how they rate
the current state of the docs. We received 48 responses, which we'll use to
focus our documentation efforts and help us improve in key areas.

Check out the results!

## Key takeaways

- Folks want more visual aids like diagrams and screenshots.
- Of the three types of documentation we asked about (component concepts,
  installation instructions, and troubleshooting), the troubleshooting docs need
  the most work.
- When asked what information they’d most like to see added to OTel’s docs, the
  top answers were more examples and more coverage, both in depth and breadth.

> - The Collector docs were the most consulted docs set based on number of

Reviewer 2: ```suggestion
- The [Collector docs](/docs/collector/) were the most consulted docs set based on number of
```

Do you want to add a sentence that this is also what we see in our google analytics numbers? https://lookerstudio.google.com/s/tSTKxK1ECeU

Reviewer: And maybe cross-reference to the corresponding section of your year-in-review post?

- https://deploy-preview-5775--opentelemetry.netlify.app/blog/2024/year-in-review/#which-pages-were-the-most-popular
- Which would correspond to the path `../year-in-review/#which-pages-were-the-most-popular`

  ratings.

> - The Java documentation was the overall highest rated docs, after normalization

Reviewer 2: ```suggestion
- The [Java documentation](/docs/languages/java/) was the overall highest rated docs, after normalization
```

looks like the work put into it is honored, cc @zeitlinger @jeanbisutti @jack-berg @trask @jaydeluca

Reviewer: Very nice! Might again be worth cross-referencing our two posts.

>   and weighting, while the Swift docs received the overall lowest rating

Reviewer 2: ```suggestion
  and weighting, while the [Swift](/docs/languages/swift/) docs received the overall lowest rating
```

> - Among the six most popular docs sets, the JavaScript docs received the lowest

Reviewer 2: ```suggestion
- Among the six most popular docs sets, the [JavaScript](/docs/languages/js/) docs received the lowest
```

---
title: Insights from the OpenTelemetry Docs Usability Survey
linkTitle: OTel-docs Survey
date: 2024-12-13
author: '[Tiffany Hrabusa](https://github.com/tiffany76) (Grafana Labs)'
issue: https://github.com/open-telemetry/opentelemetry.io/issues/5793
sig: Communications, End-User
cSpell:ignore: Hrabusa
---

[The OpenTelemetry End-User SIG](/community/end-user/) recently surveyed the
community to find out how user-friendly [OpenTelemetry's documentation](/docs/)
is. In an earlier survey, two-thirds of respondents named comprehensive
documentation as a top resource they wished they'd had when getting started with
OpenTelemetry. So we decided to dig a little deeper. The Docs Usability Survey
asked users where they go for OTel documentation, what they'd like to see more
of in the docs, and how they rate the current state of the docs. We received 48
responses, which we'll use to focus our documentation efforts and help us
improve in key areas.

Check out the results!

## Key takeaways

> - Folks want more visual aids like diagrams and screenshots.
> - Of the three types of documentation we asked about (component concepts,
>   installation instructions, and troubleshooting), the troubleshooting docs need
>   the most work.
> - When asked what information they'd most like to see added to OTel's docs, the
>   top answers were **more examples** and **more coverage**, both in depth and
>   breadth.
> - The Collector docs were the most consulted docs set based on number of
>   ratings. This result tallies with the page view analysis in the SIG
>   Communications'
>   [year-end review](../year-in-review/#which-pages-were-the-most-popular).
> - After normalization and weighting, the
>   [Java documentation](/docs/languages/java/) was the overall highest rated docs
>   set, suggesting that the efforts made to
>   [improve its information architecture](../year-in-review/#ia-improvements)
>   have paid off. Meanwhile, the [Swift](/docs/languages/swift/) docs received
>   the overall lowest rating.
> - Among the six most popular docs sets, the [JavaScript](/docs/languages/js/)
>   docs received the lowest rating.

Reviewer: How about the following:

```suggestion
- Respondents expressed a desire for **more visual aids**, such as diagrams and
  screenshots.
- Of the three types of documentation we asked about (component concepts,
  installation instructions, and troubleshooting), the **troubleshooting docs**
  were identified as needing the most improvement.
- When asked about the information they’d most like to see added to OTel's docs,
  the top responses were **more examples** and **expanded coverage**, both in
  depth and breadth.
- The [Collector](/docs/collector/) docs emerged as the most frequently
  consulted resource, a finding that aligns with the page view analysis in the
  SIG Communications'
  [year-end review](../year-in-review/#which-pages-were-the-most-popular).
- After normalization and weighting, the [Java](/docs/languages/java/)
  documentation received the **highest overall rating**, reflecting the positive
  impact of recent improvements to its
  [organization](../year-in-review/#ia-improvements). Conversely, the
  [Swift](/docs/languages/swift/) docs received the lowest overall rating.
- Among the six most popular documentation sets, the
  [JavaScript](/docs/languages/js/) docs received the **lowest rating**.
```

Author: If the highest ratings are takeaways, I think the lowest ones are too. I think we should keep them, but I'm also fine removing them if you think we should.

## Detailed insights

### About the respondents

- 79% are using OTel in production.
- 21% work for an observability or APM vendor.
- 98% have previous knowledge of observability: intermediate (60%) or expert
  (38%).

#### Q: What source do you primarily rely on when you're looking for information about OpenTelemetry?

- Overall, the majority of respondents (52%) rely on the opentelemetry.io
  documentation.
- Respondents early in their observability practice (beginner and intermediate)
  are more likely to use the opentelemetry.io documentation.
- Expert observability practitioners prefer the code repository documentation.

> ![Percentage of website users by observability knowledge level](website-usage.png)

Reviewer: How about replacing this image with markdown link this (it makes for a better viewing experience in dark mode):

```suggestion
> Respondents who use [opentelemetry.io] as their
> primary information source<br>_By level of observability knowledge_
>
> | **Beginners** | **Intermediates** | **Experts** |
> | :-----------: | :---------------: | :---------: |
> |     100%      |        62%        |     44%     |
```

Here's what it looks like:

> <img width="800" alt="image" src="https://github.com/user-attachments/assets/bfe1fbfb-7abc-4a9c-8cb8-0b4336d12775" />

Reviewer: If you agree, then pls drop the corresponding image.

Author: Thanks. I didn't think about dark mode. I'll make the change.

Most respondents believe that this section of docs needs work. Only 15% rated
the troubleshooting docs 4 or 5, and they were all intermediate-level
respondents. None of the expert-level respondents rated the troubleshooting docs
above a 3.

![Stacked row chart showing the percentage of users who rated the troubleshooting documentation from 1 (low) to 5 (high)](troubleshooting.png)

#### Q: How would you rate your experience using the current OTel documentation for the following languages and components?

Respondents were asked to rate only the documentation that applied to them, so
we can infer based on their responses which docs sets are the most used.

- The Collector documentation is the most used: 77% of respondents rated it.
- The next five documentation sets are close in popularity, with 50 to 67% of
  respondents rating them.

![Horizontal bar graph showing the number of ratings per OTel component](top-six-bar.png)

Here are the tabulated ratings for all languages and components. When the
results are normalized and weighted, we can see additional insights:

- The Java documentation has the highest overall rating.
- The Swift documentation has the lowest overall rating.

> ![Table showing the number of ratings (poor, okay, great) by OpenTelemetry documentation set](ratings-table.png)

Reviewer: How about:

```suggestion
> **How would you rate your experience using the current OTel documentation for
> the following languages and components?**
>
> | Language or component | Poor | Okay | Great | Total responses | Normalized & weighted |
> | :-------------------- | :--: | :--: | :---: | :-------------: | :-------------------: |
> | .NET                  |  4   |  8   |   2   |       14        |        5.7143         |
> | C++                   |  0   |  7   |   0   |        7        |        6.0000         |
> | Erlang                |  1   |  6   |   0   |        7        |        5.4286         |
> | GO                    |  6   |  12  |   9   |       27        |        7.1111         |
> | Java                  |  3   |  16  |   8   |       27        |      **7.3333**       |
> | JavaScript            |  3   |  19  |   2   |       24        |        6.0000         |
> | PHP                   |  1   |  4   |   2   |        7        |        7.1429         |
> | Python                |  6   |  17  |   8   |       31        |        6.7742         |
> | Ruby                  |  1   |  5   |   1   |        7        |        6.2857         |
> | Rust                  |  4   |  4   |   2   |       10        |        5.6000         |
> | Swift                 |  3   |  3   |   0   |        6        |      **4.0000**       |
> | Collector             |  9   |  17  |  11   |       37        |        6.8108         |
> | Kubernetes            |  6   |  20  |   6   |       32        |        6.3750         |
> | FaaS                  |  5   |  7   |   0   |       12        |        4.3333         |
> | **Total**             |  52  | 145  |  51   |                 |                       |
```

I think that we should sort by the value of the final column, like this:

```suggestion
> **How would you rate your experience using the current OTel documentation for the following languages and components?**
>
> | Language or component | Poor | Okay | Great | Total responses | Normalized & weighted |
> |:-----------------------|:----:|:----:|:-----:|:---------------:|:----------------------:|
> | Java                  | 3    | 16   | 8     | 27              | 7.3333                |
> | PHP                   | 1    | 4    | 2     | 7               | 7.1429                |
> | GO                    | 6    | 12   | 9     | 27              | 7.1111                |
> | Collector             | 9    | 17   | 11    | 37              | 6.8108                |
> | Python                | 6    | 17   | 8     | 31              | 6.7742                |
> | Kubernetes            | 6    | 20   | 6     | 32              | 6.3750                |
> | C++                   | 0    | 7    | 0     | 7               | 6.0000                |
> | JavaScript            | 3    | 19   | 2     | 24              | 6.0000                |
> | Ruby                  | 1    | 5    | 1     | 7               | 6.2857                |
> | Rust                  | 4    | 4    | 2     | 10              | 5.6000                |
> | .NET                  | 4    | 8    | 2     | 14              | 5.7143                |
> | Erlang                | 1    | 6    | 0     | 7               | 5.4286                |
> | FaaS                  | 5    | 7    | 0     | 12              | 4.3333                |
> | Swift                 | 3    | 3    | 0     | 6               | 4.0000                |
> | **Total**             | 52   | 145  | 51    |                 |                      |
```

If we combine these insights, we can see that the documentation used by the most
people that needs the most work is the JavaScript documentation.

> ![Subset of the previous table showing the ratings for the six most popular documentation sets](javascript-table.png)

Reviewer: Again, how about using a markdown table here?

```suggestion
> **Where should we focus our improvement efforts?**<br> _JavaScript is one of
> the six most-used docs sets, but its rating is the lowest._
>
> | Language or component | Poor | Okay | Great | Total responses | Normalized & weighted |
> | :-------------------- | :--: | :--: | :---: | :-------------: | :-------------------: |
> | GO                    |  6   |  12  |   9   |       27        |        7.1111         |
> | Java                  |  3   |  16  |   8   |       27        |        7.3333         |
> | JavaScript            |  3   |  19  |   2   |       24        |        6.0000         |
> | Python                |  6   |  17  |   8   |       31        |        6.7742         |
> | Collector             |  9   |  17  |  11   |       37        |        6.8108         |
> | Kubernetes            |  6   |  20  |   6   |       32        |        6.3750         |
```

If you agree, drop the corresponding image. Here's what it looks like:

<img width="664" alt="image" src="https://github.com/user-attachments/assets/b9ade371-e1a9-4b9a-9cb1-af87d405d284" />

---
title: Insights from the OpenTelemetry Docs Usability Survey
linkTitle: OTel-docs Survey

> date: 2024-12-13

Reviewer: ```suggestion
date: 2024-12-18
```
